griptape/drivers/vector/pgvector_vector_store_driver.py

```diff
@@ -148,8 +149,19 @@ def query(
             # The query should return both the vector and the distance metric score.
             query_result = session.query(self._model, op(vector).label("score")).order_by(op(vector))  # pyright: ignore
 
+            filter_kwargs: Optional[OrderedDict] = None
+
             if namespace:
-                query_result = query_result.filter_by(namespace=namespace)
+                filter_kwargs = OrderedDict(namespace=namespace)
+
+            if "filter" in kwargs and isinstance(kwargs["filter"], dict):
+                if filter_kwargs:
+                    filter_kwargs.update(kwargs["filter"])
+                else:
+                    filter_kwargs = OrderedDict(kwargs["filter"])
+
+            if filter_kwargs:
+                query_result = query_result.filter_by(**filter_kwargs)
```

Comment:
Can we simplify this entire thing to something like:

```python
        if "filter" in kwargs and isinstance(kwargs["filter"], dict):
                query_result.filter_by(OrderedDict(kwargs["filter"]))
```

Reply:
I want the `namespace` to be added to the OrderedDict first if it is present so that it is applied first. But I simplified this logic with the most recent commit update, so please take a look.